### ISSUE TITLE
allow pin Poetry

### ISSUE THREAD

**Issue by Borda** [pre-commit/pre-commit#2730]
### search you tried in the issue tracker

Poetry freeze

### describe your issue

Recently there was an issue of miss-compatibility between Poetry (which is used under the hood of pre-commit) and Isort (formation hook). It would be very helpful if we could also configure the Poetry version and follow the auto-update strategy with bumping versions, as it is already with all hooks. This would prevent sudden/unexpected crashes... :chipmunk: 

seems like suddenly `poetry` does not like `isort` :(
ref: https://github.com/home-assistant/core/issues/86892
also: [stackoverflow.com/a/75269701/4521646](https://stackoverflow.com/a/75269701/4521646)

In this case, it was resolved quite fast, but there can easily be a situation in that hooks won't be compatible for days, and in such case, pre-commit is completely useless and need to be disabled (or am I wrong?)

### pre-commit --version

3.0.2

### .pre-commit-config.yaml

```yaml
default_language_version:
  python: python3

ci:
  autofix_prs: true
  autoupdate_commit_msg: "[pre-commit.ci] pre-commit suggestions"
  autoupdate_schedule: quarterly
  # submodules: true

repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.4.0
    hooks:
      - id: end-of-file-fixer
      - id: trailing-whitespace
      # - id: check-json
      - id: check-yaml
      - id: check-toml
      - id: check-docstring-first
      - id: check-executables-have-shebangs
      - id: check-case-conflict
      - id: check-added-large-files
        args: ["--maxkb=100", "--enforce-all"]
      - id: detect-private-key

  - repo: https://github.com/asottile/pyupgrade
    rev: v3.3.1
    hooks:
      - id: pyupgrade
        args: [--py36-plus]
        name: Upgrade code

  - repo: https://github.com/PyCQA/docformatter
    rev: v1.5.1
    hooks:
      - id: docformatter
        args: [--in-place, --wrap-summaries=115, --wrap-descriptions=120]

  - repo: https://github.com/PyCQA/isort
    rev: 5.11.4
    hooks:
      - id: isort
        name: imports
        require_serial: false

  - repo: https://github.com/psf/black
    rev: 22.12.0
    hooks:
      - id: black
        name: Format code

  - repo: https://github.com/executablebooks/mdformat
    rev: 0.7.16
    hooks:
      - id: mdformat
        additional_dependencies:
          - mdformat-gfm
          - mdformat-black
          - mdformat_frontmatter
        exclude: |
          (?x)^(
              CHANGELOG.md|
              docs/paper_JOSS/paper.md
          )$

  - repo: https://github.com/asottile/yesqa
    rev: v1.4.0
    hooks:
      - id: yesqa
        additional_dependencies:
          - pep8-naming

  - repo: https://github.com/charliermarsh/ruff-pre-commit
    rev: v0.0.237
    hooks:
      - id: ruff
        args: ["--fix"]
```

### ~/.cache/pre-commit/pre-commit.log (if present)

https://results.pre-commit.ci/run/github/323721661/1675041322.jR8Cfbl4S9-masefqqyWKQ

**Comment by asottile** [comment]
there isn't really anything pre-commit can help you with here -- you could use `PIP_CONSTRAINT` or otherwise.  my recommendation is to not use poetry personally

**Comment by Borda** [comment]
> my recommendation is to not use poetry personally

so why is the bot running poetry? :thinking:

**Comment by asottile** [comment]
> > my recommendation is to not use poetry personally
> 
> so why is the bot running poetry? 🤔

because isort selected poetry

**Comment by Borda** [comment]
Ok, I think I got it, and apology for all the confusion... :rabbit:

**Comment by neilsh** [comment]
For anyone else encountering this, the issue is fixed in isort==5.12.0: 

* https://github.com/PyCQA/isort/releases/tag/5.12.0
* https://github.com/PyCQA/isort/pull/2078
